src/sage/modules/free_module.py

```diff
@@ -6701,16 +6702,15 @@ def __init__(self, ambient, basis, check=True,
         C = self.element_class
         w = [C(self, x.list(), coerce=False, copy=False) for x in basis]
         self.__basis = basis_seq(self, w)
-
-        if echelonize or already_echelonized:
-            self.__echelonized_basis = self.__basis
-        else:
-            if echelonized_basis is None:
-                echelonized_basis = self._echelonized_basis(ambient, basis)
-            w = [C(self, x.list(), coerce=False, copy=True)
-                 for x in echelonized_basis]
-            self.__echelonized_basis = basis_seq(self, w)
-        if check and len(basis) != len(self.__echelonized_basis):
+        # This is needed but for what reason or why it fails without this, I am not sure!
+        # Essentially this allows conversion from say eg Polynomial_flint to Generic one or something!
+        self.__converted_basis = basis
+
+        echelonized: bool = echelonize or already_echelonized
```

Comment:
```suggestion
        echelonized = echelonize or already_echelonized
```
Let's not add unnecessary Python type indicator code.